src/Core/Utility/Eigen.h

```diff
@@ -53,7 +53,7 @@ Eigen::Vector6d TransformMatrix4dToVector6d(const Eigen::Matrix4d &input);
 
 /// Function to solve Ax=b
 std::tuple<bool, Eigen::VectorXd> SolveLinearSystem(
-    const Eigen::MatrixXd &A, const Eigen::VectorXd &b);
+    const Eigen::MatrixXd &A, const Eigen::VectorXd &b, bool check_det = true);
```

Comment:
it checkes determinant by default. Hence, it should not affect original pipeline.